modules/internal/SharedObject.chpl

```diff
@@ -100,6 +100,7 @@ See also :ref:`about-owned-intents-and-instantiation` which includes examples.
  */
 module SharedObject {
 
+  private use ChapelError;
```

Comment:
You don't have to fix it in this PR but it seems odd that SharedObject does not `private use ChapelStandard;` before this line. Perhaps that would allow it to `private use OwnedObject`.

Reply:
I was taking the approach of (a) trying to make all `use ChapelStandard` statements private while (b) introducing the minimal number of uses required to run the test system (i.e., why introduce a bunch of symbols that aren't actually required or needed...?).  Specifically, I didn't make any attempt to make other uses private, as I was just focusing on ChapelStandard for now (the reason being that if we were searching modules redundantly along various `private use` arcs, that could result in lots of extra searching if we didn't short-circuit intelligently, which is why I was hoping making them private would improve compile-time... since it didn't, that suggests maybe we are short-circuiting intelligently, or maybe it just isn't all that impactful.  But the change still seemed like a step in the right direction).

Put another way, it very well may be that we could `private use OwnedObject` here as well, I just didn't try that because it wasn't what I was exploring and wasn't necessary to make the tests pass.  It seems likely that use could be `private` since I think all of OwnedObject's `use` statements are now private themselves (?).

Reply:
(@mppf: I'm holding off on merging until I'm more sure I'm not misunderstanding you).

Reply:
oh, you're good to go, we agree it is future work for another PR.

Reply:
But do we disagree that SharedObject should `private use ChapelStandard`?  I don't think it should since it doesn't seem to be necessary: the only modules that it uses `private use ChapelStandard`, making me think that it's got all the dependences it needs (?).

Reply:
Whatever works is fine with me. (Historically, I thought we were trying to `private use ChapelStandard` in all standard/internal modules that are included by default, but I don't know why we were doing that, or if I even understood the convention).

Reply:
Could you be thinking of this: at some point we changed the compiler so that `use ChapelStandard` would not be added by default to internal modules as it traditionally had been—only to user and standard modules.  As a result of that change, we added a bunch of `use ChapelStandard` lines to internal modules to keep them working when they'd been depending on various standard features.  Better might be to replace these "big hammers" with the more precise `use`s that the module really requires (though maybe that's too fiddly to be worthwhile).

If that wasn't it, I don't remember any other reason that we would've decided to `use ChapelStandard` in all included-by-default internal modules.